pycbc/psd/variation.py

```diff
@@ -160,3 +165,220 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+    segment : {float, 8}
+        Duration of the segments for the mean square estimation in seconds.
+    short_segment : {float, 0.25}
+        Duration of the short segments for the outliers removal.
+    psd_long_segment : {float, 512}
+        Duration of the long segments for PSD estimation in seconds.
+    psd_duration : {float, 16}
+        Duration of the segments for PSD estimation in seconds.
+    psd_stride : {float, 8}
+        Separation between PSD estimation segments in seconds.
+    psd_avg_method : {string, 'median'}
+        Method for averaging PSD estimation segments.
+    low_freq : {float, 20}
+        Minimum frequency to consider the comparison between PSDs.
+    high_freq : {float, 480}
+        Maximum frequency to consider the comparison between PSDs.
+
+    Returns
+    -------
+    psd_var : TimeSeries
+        Time series of the variability in the PSD estimation
+    """
+    # Calculate strain precision
+    if strain.precision == 'single':
+        fs_dtype = numpy.float32
+    elif strain.precision == 'double':
+        fs_dtype = numpy.float64
+
+    # Convert start and end times immediately to floats
+    start_time = numpy.float(strain.start_time)
+    end_time = numpy.float(strain.end_time)
+
+    # Resample the data
+    strain = resample_to_delta_t(strain, 1.0/2048)
+    srate = int(strain.sample_rate)
+
+    # Fix the step for the PSD estimation and the time to remove at the
+    # edge of the time series.
+    step = 1.0
+    strain_crop = 8.0
+
+    # Find the times of the long segments
+    times_long = numpy.arange(start_time, end_time,
+                              psd_long_segment-2*strain_crop
+                              - segment + step)
+
+    # Set up the empty time series for the PSD variation estimate
+    psd_var = TimeSeries(zeros(int(numpy.floor((end_time - start_time -
+                                                2*strain_crop - segment +
+                                                1) / step))),
+                         delta_t=step, copy=False,
+                         epoch=start_time + strain_crop + segment)
+
+    # Create a bandpass filter between low_freq and high_freq
+    filt = sig.firwin(4*srate, [low_freq, high_freq], pass_zero=False,
+                      window='hann', nyq=srate/2)
+    filt.resize(int(psd_duration*srate))
+    # Fourier transform the filter and take the absolute value to get
+    # rid of the phase. Save the filter as a frequency series.
+    filt = abs(rfft(filt))
+    my_filter = FrequencySeries(filt, delta_f=1./psd_duration,
+                                dtype=fs_dtype)
+
+    ind = 0
+    for tlong in times_long:
+        # Calculate PSD for long segment
+        if tlong + psd_long_segment <= float(end_time):
+            astrain = strain.time_slice(tlong, tlong + psd_long_segment)
+            plong = pycbc.psd.welch(astrain,
+                                    seg_len=int(psd_duration
```

Comment:
put `astrain` on a new line which is indented less so that the `seg_len` assignments don't need to be split over lines